monkey/infection_monkey/master/mock_master.py

```diff
@@ -0,0 +1,112 @@
+import logging
+
+from infection_monkey.i_master import IMaster
+from infection_monkey.i_puppet import IPuppet, PortStatus
+from infection_monkey.model.host import VictimHost
+from infection_monkey.telemetry.exploit_telem import ExploitTelem
+from infection_monkey.telemetry.file_encryption_telem import FileEncryptionTelem
+from infection_monkey.telemetry.messengers.i_telemetry_messenger import ITelemetryMessenger
+from infection_monkey.telemetry.post_breach_telem import PostBreachTelem
+from infection_monkey.telemetry.scan_telem import ScanTelem
+from infection_monkey.telemetry.system_info_telem import SystemInfoTelem
+
+logger = logging.getLogger()
+
+
+class MockMaster(IMaster):
+    def __init__(self, puppet: IPuppet, telemetry_messenger: ITelemetryMessenger):
+        self._puppet = puppet
+        self._telemetry_messenger = telemetry_messenger
+        self._hosts = {
+            "10.0.0.1": VictimHost("10.0.0.1"),
+            "10.0.0.2": VictimHost("10.0.0.2"),
+            "10.0.0.3": VictimHost("10.0.0.3"),
+            "10.0.0.4": VictimHost("10.0.0.4"),
+        }
+
+    def start(self) -> None:
+        self._run_sys_info_collectors()
+        self._run_pbas()
+        self._scan_victims()
+        self._fingerprint()
+        self._exploit()
+        self._run_payload()
+
+    def _run_sys_info_collectors(self):
+        system_info_telemetry = {}
+        system_info_telemetry["ProcessListCollector"] = self._puppet.run_sys_info_collector(
+            "ProcessListCollector"
+        )
+        self._telemetry_messenger.send_telemetry(
+            SystemInfoTelem({"collectors": system_info_telemetry})
+        )
+        system_info = self._puppet.run_sys_info_collector("LinuxInfoCollector")
+        self._telemetry_messenger.send_telemetry(SystemInfoTelem(system_info))
+
+    def _run_pbas(self):
+        name = "AccountDiscovery"
+        command, result = self._puppet.run_pba(name, {})
+        self._telemetry_messenger.send_telemetry(PostBreachTelem(name, command, result))
+
+        name = "CommunicateAsBackdoorUser"
+        command, result = self._puppet.run_pba(name, {})
+        self._telemetry_messenger.send_telemetry(PostBreachTelem(name, command, result))
+
+    def _scan_victims(self):
+        ips = ["10.0.0.1", "10.0.0.2", "10.0.0.3"]
+        ports = [22, 445, 3389, 8008]
+        for ip in ips:
+            h = self._hosts[ip]
```

Comment:
d n a. 
(it means "do not abbreviate" :D)

Reply:
For the actual implementation, sure. For the mock, I don't much care. It's probably a good idea.